### PR TITLE
CDAP-3164 add kafka and jms source docs

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/table.rst [hunk 1]
```diff
@@ -27,6 +27,8 @@ explored through Hive.
 **schema.row.field:** The name of the record field that should be used as the row
 key when writing to the table.
 
+**case.sensitive.row.field:** Whether or not 'schema.row.field' is case sensitive, defaults to true.
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/table.rst]: Whether 'schema.row.field' is case-sensitive; defaults to true.

File: cdap-docs/application-templates/source/etl/templates/sinks/real-time/table.rst [hunk 2]
```diff
@@ -27,6 +27,8 @@ explored through Hive.
 **schema.row.field:** The name of the record field that should be used as the row
 key when writing to the table.
 
+**case.sensitive.row.field:** Whether or not 'schema.row.field' is case sensitive, defaults to true.
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/real-time/table.rst]: Whether 'schema.row.field' is case-sensitive; defaults to true.

File: cdap-docs/application-templates/source/etl/templates/sources/real-time/jms.rst [hunk 3]
```diff
@@ -8,11 +8,12 @@ Sources: Real-time: JMS
 
 .. rubric:: Description
 
-JMS Real-time Source. Emits a record with a field 'message' of string type.
+Java Message Service (JMS) Real-time Source. Emits a record with a field 'message' of string type.
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sources/real-time/jms.rst]: of type string?

File: cdap-docs/application-templates/source/etl/templates/sources/real-time/kafka.rst [hunk 4]
```diff
@@ -39,4 +40,38 @@ If no format is given, Kafka message payloads will be treated as bytes, resultin
...
+          {\"name\":\"item\",\"type\":\"string\"},
+          {\"name\":\"count\",\"type\":\"int\"},
+          {\"name\":\"price\",\"type\":\"double\"}
+        ]
+      }"
+    }
+  }
+
+This example reads from ten partitions of the 'purchases' topic of a Kafka instance.
+It connects to Kafka via a Zookeeper instance running on 'localhost'. It also
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sources/real-time/kafka.rst]: It also -> It...the specified schema.

File: cdap-docs/application-templates/source/etl/templates/sources/real-time/kafka.rst [hunk 5]
```diff
@@ -39,4 +40,38 @@ If no format is given, Kafka message payloads will be treated as bytes, resultin
...
+parses Kafka messages using the 'csv' format into records with the given schema.
+For each Kafka message read, it will output a record with schema::
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sources/real-time/kafka.rst]: with the schema::

**Comment by albertshau** [comment]
@awholegunch addressed comments, please take another look

File: cdap-docs/application-templates/source/etl/templates/sources/real-time/kafka.rst (hunk 4 above)
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sources/real-time/kafka.rst]: OK, how about:
````
It then parses...
```
as "It also" sounds weak.

**Comment by awholegunch** [comment]
One comment, otherwise LGTM.

**Comment by henry-cask** [comment]
LGTM

**Pull request merged**